cloud/amazon/route53.py

```diff
@@ -344,6 +367,11 @@ def main():
           elif not alias_hosted_zone_id_in:
               module.fail_json(msg = "parameter 'alias_hosted_zone_id' required for alias create/delete")
 
+    if vpc_id_in and not private_zone_in:
```

Comment:
Thanks @jsmartin but I don't think that's applicable here. It's valid to specify private_zone True and not specify a vpc_id, but it's not valid to specify a vpc_id but not private_zone True. I think that required_together enforces the second use case but doesn't allow for the first, if I understand it correctly.

I think what I actually want is something similar to required_if, but I can't find any documentation about it, it doesn't seem to be used in any of ansible-modules-core or ansible-modules-extras. I also don't think that required_if quite supports the use case. Looking at https://github.com/ansible/ansible/blob/devel/lib/ansible/module_utils/basic.py#L979 I think there are two things that required_if doesn't support:
1. I want to specify that the private_zone must have a specific value, not just that is passed in (i.e., somebody could pass in private_zone: False and it would pass the required_if check)
1. I want to enforce the previous condition if and only if vpc_id is specified, while required_if only lets me enforce it if the value of vpc_id matches a specific value. It doesn't seem to support the case of "enforce the condition if my parameter is specified" (it's currently "enforce the condition if my parameter takes on a specific value").

Thoughts? I didn't see anything else in that file that looked like it would do what I want.

--Joel